### PR TITLE
golang support added

### PR DESCRIPTION
**Pull request by yzainee**
Added functionality to support the golang vuln ingestion.
Few new fields added to make golang vuln check via commit hash easier.

### CODE REVIEW

**Review by jparsai — APPROVED** [review]

File: src/cve.py [hunk 1]
```diff
@@ -64,7 +64,14 @@ def create_pv_nodes(self):
         else:
             logger.info("Latest version is affected {p} {v}".format(p=p, v=latest_version))
 
-        for ver in self._snyk_pkg_data.get('affected'):
+        if e == 'golang':
+            itr_list = self._snyk_pkg_data.get('all_ver')
+            epv_dict['gh_link'] = self._snyk_pkg_data.get('gh_link')
+            epv_dict['license'] = self._snyk_pkg_data.get('license')
+        else:
+            itr_list = self._snyk_pkg_data.get('affected')
```
Comment (prashbnair) [inline src/cve.py:72]: Has Snyk got a different name for the 'affected' field in case of golang?

File: src/cve.py [hunk 2]
```diff
@@ -153,6 +160,14 @@ def get_qstring_for_cve_node(self, vulnerability):
                 title = re.sub("[\'\"]", "", ref.get('title'))
                 ref_str = title + ":" + ref.get('url')
                 query_str += "cve_v.property('references', '" + ref_str + "');"
+        
+        if vulnerability.get('ecosystem') == 'golang':
```
Comment (prashbnair) [inline src/cve.py:164]: This looks fine for now, as we add more languages if these conditions keep getting added we will need to look at some kind of inheritance which can handle this.

**Review by prashbnair — APPROVED** [review]

File: src/cve.py (hunk 1 above)
Comment (yzainee) [reply src/cve.py:72]: For other ecosystems, we pick only the affected versions and create those while ingesting vuln and rely on the upstream monitoring to take care of all other versions. As we dont have upstream monitoring for golang, here i am taking all the versions available for a pkg and then creating them.
Note: the edges between vuln and the affected versions will remain the same as in other ecosystems.

File: src/cve.py (hunk 2 above)
Comment (yzainee) [reply src/cve.py:164]: This condition made sense only for golang as no other ecosystem supports commit hashes. To flag vuln for commit hash as well, i had to add this condition here

**Pull request merged**